src/Chapter4.hs

```diff
@@ -611,6 +630,19 @@ Implement the 'Monad' instance for our lists.
   maybe a few) to flatten lists of lists to a single list.
 -}
 
+instance Monad List where
+  (>>=) :: List a -> (a -> List b) -> List b
+  Empty >>= _ = Empty
+  Cons a as >>= f = go (f a) (as >>= f)
+    where go :: List a -> List a -> List a
+          go Empty bs = bs
+          go (Cons a as) bs = Cons a (go as bs)
+
+replicate' :: Int -> a -> List a
+replicate' n x
+  | n == 1 = Cons x Empty
+  | n > 1 = Cons x (replicate' (n - 1) x)
+  | otherwise = Empty
```

Comment:
This function would be a great example of using the recursive `go` pattern! In the beginning, you can check all invalid inputs. E.g. if the number is negative and return an empty list immediately. And then, in the recursive `go` you can assume that it's possible and implement a simple recursive function with two pattern cases.